improver/wxcode/weather_symbols.py

```diff
@@ -491,17 +403,17 @@ def _constraint_string(diagnostic, threshold_name, threshold_val):
             elif diagnostic in self.threshold_coord_names:
                 threshold_coord_name = self.threshold_coord_names[diagnostic]
             else:
-                threshold_coord_name = get_threshold_coord_name_from_probability_name(
+                threshold_coord_name = get_diagnostic_cube_name_from_probability_name(
```

Comment:
Again - I don't think this should have changed and I'm confused by why the unit tests pass, since the threshold coordinate name and diagnostic name are different for "in vicinity" diagnostics.

It's possible we need to update the unit test setup utilities to set the correct metadata for "in vicinity" cubes so that we have a test to cover this difference.  Perhaps I should also review the acceptance test inputs.